### PR TITLE
Add Fréchet Radiomics Distance (FRD) to MONAI metrics (#8643)

### LINKED ISSUE #8643: Add Fréchet Radiomics Distance (FRD-Score) to MONAI Metrics
**Is your feature request related to a problem? Please describe.**
Many existing metrics for evaluating generative, reconstruction, and harmonization pipelines in medical imaging, e.g. those based on neural representations, lack interpretability, are modality-dependent, and often fail to capture clinically meaningful image characteristics. We conducted some recent work proposing and empirically validating the Fréchet Radiomics Distance (FRD) [[1](https://arxiv.org/abs/2412.01496),[2](https://arxiv.org/abs/2403.13890)] showing that, unlike [FID](https://monai-dev.readthedocs.io/en/fixes-sphinx/metrics.html#frechet-inception-distance), FRD can directly be used for both 2D and 3D images, be conditioned by anatomical masks, is robust to small sample sizes, and correlates more strongly with clinical utility by reflecting anatomically and biomarker-relevant differences. These advantages address key limitations of current metrics (e.g. FID) and support standardized, reproducible comparison of real, synthetic, and cross-domain medical-imaging datasets.

**Describe the solution you'd like**
We maintain the open-source FRD-Score repository [[3](github.com/RichardObi/frd-score)] and propose to integrate it into MONAI by refactoring the code into a MONAI-conformant metric class (e.g., `FrechetRadiomicsDistance` in `monai.metrics`) that:
- Extracts radiomic feature vectors for two image sets (based on [PyRadiomics](https://pyradiomics.readthedocs.io/en/latest/))
- Computes the Fréchet distance between the resulting feature distributions
- Integrates with MONAI’s existing metric pipeline

**Describe alternatives you've considered**
We evaluated embedding-based image distribution metrics such as FID and deep-feature distances [1](https://arxiv.org/abs/2412.01496), but these depend on pre-trained networks whose feature spaces may not reflect medical-imaging semantics.

**Additional context**
[1] Konz, N., Osuala, R., Verma, P., Chen, Y., Gu, H., Dong, H., Chen, Y., Marshall, A., Garrucho, L., Kushibar, K., Lang, D. M., et al. “Fréchet Radiomic Distance (FRD): A Versatile Metric for Comparing Medical Imaging Datasets.” arXiv preprint arXiv:2412.01496. (Comment: Accepted in Medical Image Analysis, subject to minor revisions) [https://arxiv.org/abs/2412.01496](https://arxiv.org/abs/2412.01496)
[2] Osuala, R., Lang, D. M., Verma, P., Joshi, S., Tsirikoglou, A., Skorupko, G., Kushibar, K., et al.
“Towards Learning Contrast Kinetics with Multi-Condition Latent Diffusion Models.” In: MICCAI 2024, pp. 713–723. Springer Nature Switzerland. arXiv:2403.13890. [https://arxiv.org/abs/2403.13890](https://arxiv.org/abs/2403.13890)
[3] Official implementation of the Fréchet Radiomic Distance (FRD). [https://github.com/RichardObi/frd-score](https://github.com/RichardObi/frd-score)
Image

### PR DESCRIPTION
**Pull request by AymanL**
Fixes #8643

### Description
Implements FRD as in Konz et al. (arXiv:2412.01496): same Fréchet distance formula as FID, applied to pre-extracted radiomic features. 
Supports 2D/3D and optional mask conditioning via external feature extraction (e.g. PyRadiomics).

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by ericspod** [comment]
Hi @AymanL thanks for this contribution. It seems that this metric is equivalent to `FIDMetric` (as shown in your tests) but is used for comparing radiomic values rather than images. I get from the description and the paper why this is done but it's less clear why a separate class is needed for this. To provide enough material to know how to use this class, you need to provide a tutorial [here](https://github.com/Project-MONAI/tutorials) demonstrating how to extract radiomic values and compute the FRD, in which case you could just use `FIDMetric`. 

Also I see in the docstring "The same Fréchet distance formula as in FID is applied to the mean and covariance of these features." Does this imply that there should be some other computation being done that is different from what `FIDMetric` does? 

I think the Coderabbit comment is worth responding to with a more precise check of the input values, I know `get_fid_score` will raise an exception if the dimensions are wrong but a more descriptive error before that happens would help users.
